### ISSUE TITLE
Paging past max results results in query error

### ISSUE THREAD

**Issue by maz0717** [yiisoft/yii2-sphinx#95]
### What steps will reproduce the problem?
Access a page past max results

### What's expected?
Return no results.  Mysql will quietly return no rows, Sphinx won't.

### What do you get instead?
Query error

### Additional info
I tried to fix this and submit a pull request, but it seems that pagination is not initialized until after the query is executed.  According to the pagination docs, only totalCount and pageCount rely on the query being prepared first.  I'm not sure if that's a bug in pagination or the way I'm using it.  Also, max results would have to be a class var to properly implement this.

I've implemented a work around for this in my search model, right after initializing the data provider.

```
$page = isset($params['page']) ? $params['page'] : 1;
if((50 * ($page-1)) > 10000) {
    return [];
}
```

| Q                     | A
| --------------------- | ---
| Yii version           | latest
| Yii Sphinx version    | latest
| Sphinx server version | 2.2.11
| PHP version           | 7.0.27
| Operating system      | MacOS 10.13.3 and CentOS 7

**Comment by klimov-paul** [comment]
> Query error

Please, provide the full stack trace for the error you see.

**Comment by maz0717** [comment]
As requested:

```
2018-02-03 15:12:41
[157.55.39.167][-][inqm2c7ugq86k5e0p9c08vigp6][error][yii\db\Exception]
PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064
offset out of bounds (offset=20050, max_matches=10000) in
.../vendor/yiisoft/yii2/db/Command.php:1067
Stack trace:
#0
.../vendor/yiisoft/yii2/db/Command.php(1067):
PDOStatement->execute()
#1
.../vendor/yiisoft/yii2/db/Command.php(373):
yii\db\Command->queryInternal('')
#2
.../vendor/yiisoft/yii2-sphinx/Query.php(222):
yii\db\Command->query()
#3
.../vendor/yiisoft/yii2-sphinx/ActiveDataProvider.php(159):
yii\sphinx\Query->search(NULL)
#4
.../vendor/yiisoft/yii2/data/BaseDataProvider.php(101):
yii\sphinx\ActiveDataProvider->prepareModels()
#5
.../vendor/yiisoft/yii2/data/BaseDataProvider.php(114):
yii\data\BaseDataProvider->prepare()
#6
.../frontend/controllers/InventoryController.php(96):
yii\data\BaseDataProvider->getModels()
#7
.../frontend/controllers/InventoryController.php(60):
frontend\controllers\InventoryController->browseResults()
#8 [internal function]:
frontend\controllers\InventoryController->actionRecent('30')
#9
.../vendor/yiisoft/yii2/base/InlineAction.php(57):
call_user_func_array(Array, Array)
#10
.../vendor/yiisoft/yii2/base/Controller.php(157):
yii\base\InlineAction->runWithParams(Array)
#11
.../vendor/yiisoft/yii2/base/Module.php(528):
yii\base\Controller->runAction('recent', Array)
#12
.../vendor/yiisoft/yii2/web/Application.php(103):
yii\base\Module->runAction('inventory/recen...', Array)
#13
.../vendor/yiisoft/yii2/base/Application.php(386):
yii\web\Application->handleRequest(Object(yii\web\Request))
#14 .../frontend/web/index.php(18):
yii\base\Application->run()
#15 {main}

Next yii\db\Exception: SQLSTATE[42000]: Syntax error or access
violation: 1064 offset out of bounds (offset=20050, max_matches=10000)
The SQL being executed was: SELECT `id`, `website_publish_date`,
`acquisition_date`, weight() as relevance FROM `raremaps` WHERE
(`website_publish_date` >= 1515078761) AND (`website_publish_date` <=
1517670761) AND (`website_status_id`=2) ORDER BY
`website_publish_date` DESC LIMIT 20050,50 OPTION max_matches = 10000;
SHOW META in
.../gsys/vendor/yiisoft/yii2-sphinx/Schema.php:589
Stack trace:
#0
.../vendor/yiisoft/yii2/db/Command.php(1082):
yii\sphinx\Schema->convertException(Object(PDOException), 'SELECT
`id`, `w...')
#1
.../vendor/yiisoft/yii2/db/Command.php(373):
yii\db\Command->queryInternal('')
#2
.../vendor/yiisoft/yii2-sphinx/Query.php(222):
yii\db\Command->query()
#3
.../vendor/yiisoft/yii2-sphinx/ActiveDataProvider.php(159):
yii\sphinx\Query->search(NULL)
#4
.../vendor/yiisoft/yii2/data/BaseDataProvider.php(101):
yii\sphinx\ActiveDataProvider->prepareModels()
#5
.../vendor/yiisoft/yii2/data/BaseDataProvider.php(114):
yii\data\BaseDataProvider->prepare()
#6
.../frontend/controllers/InventoryController.php(96):
yii\data\BaseDataProvider->getModels()
#7
../frontend/controllers/InventoryController.php(60):
frontend\controllers\InventoryController->browseResults()
#8 [internal function]:
frontend\controllers\InventoryController->actionRecent('30')
#9
.../vendor/yiisoft/yii2/base/InlineAction.php(57):
call_user_func_array(Array, Array)
#10
.../gsys/vendor/yiisoft/yii2/base/Controller.php(157):
yii\base\InlineAction->runWithParams(Array)
#11
.../gsys/vendor/yiisoft/yii2/base/Module.php(528):
yii\base\Controller->runAction('recent', Array)
#12
.../gsys/vendor/yiisoft/yii2/web/Application.php(103):
yii\base\Module->runAction('inventory/recen...', Array)
#13
.../gsys/vendor/yiisoft/yii2/base/Application.php(386):
yii\web\Application->handleRequest(Object(yii\web\Request))
#14 /sites/beta.raremaps.com/gsys/frontend/web/index.php(18):
yii\base\Application->run()
#15 {main}
```

**Comment by klimov-paul** [comment]
Unable to reproduce. This was fixed long ago at #12.
There is a unit test, which verifies fix is working:
https://github.com/yiisoft/yii2-sphinx/blob/master/tests/ActiveDataProviderTest.php#L137-L162

**Comment by iluxaar** [comment]
![image](https://user-images.githubusercontent.com/15342468/78638724-9369c500-78ce-11ea-840a-479a919c044d.png)
